src/main/java/org/craftercms/engine/model/DefaultSiteItem.java

```diff
@@ -36,14 +40,14 @@
  */
 public class DefaultSiteItem extends AbstractXmlSiteItem {
 
-    protected Item item;
+    protected ImmutableItem item;
```

Comment:
I think there is a misunderstanding. We want to return a copy of the Item as Item. There is no need to make it immutable.

Reply:
@sumerjabri I have updated this to return a copy of the Item